src/adagrams.js

```diff
@@ -1,15 +1,119 @@
+const letterPool = {
+  /** first number in the values represents the letters frequency,
+   * the second represents its point value
+   */
+  A: [9, 1],
+  B: [2, 3],
+  C: [2, 3],
+  D: [4, 2],
+  E: [12, 1],
+  F: [2, 4],
+  G: [3, 2],
+  H: [2, 4],
+  I: [9, 1],
+  J: [1, 8],
+  K: [1, 5],
+  L: [4, 1],
+  M: [2, 3],
+  N: [6, 1],
+  O: [8, 1],
+  P: [2, 3],
+  Q: [1, 10],
+  R: [6, 1],
+  S: [4, 1],
+  T: [6, 1],
+  U: [4, 1],
+  V: [2, 4],
+  W: [2, 4],
+  X: [1, 8],
+  Y: [2, 4],
+  Z: [1, 10],
+};
+
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  /** creates a list based on the probablity of each number from the letter pool
+   * returns 10 random letters from that list */
+  let letter_list = [];
+  let returned_letters = [];
+
+  for (let letter in letterPool) {
+    for (let i = 0; i < letterPool[letter][0]; i++) {
+      letter_list.push(letter);
+    }
+  }
+
+  for (let i = 0; i < 10; i++) {
+    let current_letter =
+      letter_list[Math.floor(Math.random() * (letter_list.length - 1))];
+    let index = letter_list.indexOf(current_letter);
+    returned_letters.push(current_letter);
+    letter_list.splice(index, 1);
```

Comment:
👍 nice! this works just fine! an alternative strategy is to turn the randomly selected index value to `undefined` or a blank string once you've picked it. Then simply check if the next randomly selected index value is blank.

This might be a slightly more efficient suggestion, but it's a small change